src/sentry/api/issue_search.py

```diff
@@ -126,6 +131,21 @@ def convert_first_release_value(
     return list(releases)
 
 
+def convert_substatus_value(
+    value: Iterable[Union[str, int]],
+    projects: Sequence[Project],
+    user: User,
+    environments: Optional[Sequence[Environment]],
+) -> List[int]:
```

Comment:
```suggestion
def convert_substatus_value(
    value: Iterable[str | int],
    projects: Sequence[Project],
    user: User,
    environments: Sequence[Environment] | None,
) -> list[int]:
```

Supposedly [the new-fangled way is preferred](https://typing.readthedocs.io/en/latest/source/best_practices.html#stylistic-practices), though it's making me wonder - do you need a `=None` on `environments` if you use `| None` instead of `Optional`? The docs don't say so (and I know in JS you wouldn't) but do you in Python?

Reply:
I think I agree with [this](https://stackoverflow.com/a/73591412) answer - the `Optional` typing + default argument is easier to read than `X | None = None`

Reply:
I personally like `Optional` vs `type1 | type2`, but I've seen it done both ways throughout the codebase.